### PR TITLE
[Tests] ES6: Use Destructuring Assignment with the Rest Operator

### LINKED ISSUE #18190: Wrong answer passes: Use Destructuring Assignment with the Rest Operator to Reassign Array Elements
#### Challenge can be passed using global variable instead of array passed as parameter
There is no test to dissuade using the global variable `source` instead of the argument `list`.

```javascript
const source = [1,2,3,4,5,6,7,8,9,10];

function removeFirstTwo(list) {
  // change code below this line
  const [,, ...arr] = source; // change this - SHOULD BE list NOT source
  // change code above this line
  return arr;
}
const arr = removeFirstTwo(source);
console.log(arr); // should be [3,4,5,6,7,8,9,10]
console.log(source); // should be [1,2,3,4,5,6,7,8,9,10];
```

I think that a test confirming `removeFirstTwo([10,20,30])` returned `[30]` would be enough to prevent that.

#### 
https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/use-destructuring-assignment-with-the-rest-operator-to-reassign-array-elements/

#### Tell us about your browser and operating system

* Browser Name: Chrome 
* Browser Version: Version 69.0.3497.100 (Official Build) (64-bit)
* Operating System: Ubuntu

### PR DESCRIPTION
**Pull request by ezioda004**
Added additional tests and solution. Have tested locally.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #18190

### CODE REVIEW

**Review by nikrb — APPROVED** [review]
LGTM

**Comment by nikrb** [comment]
@ezioda004 successfully tested locally.
Just in passing ... Is requiring a specific variable name `arr` a bit, restrictive?

**Comment by ezioda004** [comment]
@nikrb It kinda is but we're providing `arr` variable in the seed anyway. Besides that, there are other challenges too which makes a particular identifier returned. Also, we already have other test that are relying on this variable. 
```
text: <code>arr</code> should be <code>[3,4,5,6,7,8,9,10]</code>
    testString: assert(arr.every((v, i) => v === i + 3) && arr.length === 8,'<code>arr</code> should be <code>[3,4,5,6,7,8,9,10]</code>');
```
So IMO its fine but if you have any suggestions we can add that.

**Comment by nikrb** [comment]
@ezioda004 yes I noticed that other test, was going to suggest it should be rewritten, but given the time constraints I figured it could wait :smile:

**Comment by nikrb** [comment]
@ezioda004 I don't see the error with #18069 on local master.
However, I think the seed should be updated because we're using the global variable in the function. I would suggest changing the seed to something like:
```
...
function removeFirstTwo(list) {
  "use strict";
  // change code below this line
  const updatedList = list; // change this
  // change code above this line
  return updatedList;
}
...
```

**Comment by ezioda004** [comment]
@nikrb Awesome, fixed.
BTW if the PR looks good to you, you can merge it.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-destructuring-assignment-with-the-rest-operator-to-reassign-array-elements.english.md [hunk 1]
```diff
@@ -63,6 +65,14 @@ console.log(source); // should be [1,2,3,4,5,6,7,8,9,10];
 <section id='solution'>
 
 ```js
-// solution required
+const source = [1,2,3,4,5,6,7,8,9,10];
+function removeFirstTwo(list) {
+  "use strict";
+  // change code below this line
+  const [, , ...arr] = list;
```
Comment (Nirajn2311) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-destructuring-assignment-with-the-rest-operator-to-reassign-array-elements.english.md]: Remove extra space:
```suggestion
  const [, , ...arr] = list;
```

**Review by Nirajn2311 — CHANGES_REQUESTED** [review]
@ezioda004 Everything LGTM, just do the below change

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-destructuring-assignment-with-the-rest-operator-to-reassign-array-elements.english.md (hunk 1 above)
Comment (ezioda004) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-destructuring-assignment-with-the-rest-operator-to-reassign-array-elements.english.md]: Fixed.

**Review by Nirajn2311 — APPROVED** [review]

**Comment by scissorsneedfoodtoo** [comment]
@ezioda004, just checked this out and everything seems to be working fine locally. The only thing I might suggest is that we consider removing the first destructuring test, the one that reads "Destructuring should be used." Seems unnecessary since we have the other test to use destructuring on list now, but having both tests wouldn't hurt. Totally up to you! Either way I think we're good to merge.

**Comment by ezioda004** [comment]
@scissorsneedfoodtoo I agree, removed the old test as it was redundant.

**Pull request merged**